src/element_handle.ts

```diff
@@ -369,4 +383,78 @@ export class ElementHandle {
       this.#page.timeout,
     );
   }
+
+  /**
+   * Executes the given function or string whose first argument is a DOM element and returns the result of the execution.
+   *
+   * @example
+   * ```ts
+   * /// <reference lib="dom" />
+   * const value: string = await element.evaluate((element: HTMLInputElement) => element.value)
+   * ```
+   *
+   * @example
+   * ```
+   * /// <reference lib="dom" />
+   * await element.evaluate(
+   *  (el: HTMLInputElement, key: string, value: string) => el.setAttribute(key, value),
+   *  { args: ["href", "astral"] }
+   * )
+   * ```
+   */
+  async evaluate<E extends Element, R extends AnyArray, T = unknown>(
+    func: ElementEvaluateFunction<E, R, T> | string,
+    evaluateOptions?: ElementEvaluateOptions<R>,
+  ): Promise<T> {
+    const { object } = await this.#celestial.DOM.resolveNode({
+      nodeId: this.#id,
+    });
+
+    const args: Runtime_CallArgument[] = [{
+      objectId: object.objectId,
+    }];
+
+    if (evaluateOptions?.args) {
+      for (const argument of evaluateOptions.args) {
+        if (Number.isNaN(argument)) {
+          args.push({
+            unserializableValue: "NaN",
+          });
+        } else {
+          args.push({
+            value: argument,
+          });
+        }
+      }
+    }
+
+    const { result, exceptionDetails } = await this.#celestial.Runtime
+      .callFunctionOn({
+        functionDeclaration: func.toString(),
+        objectId: object.objectId,
+        arguments: args,
+        awaitPromise: true,
+        returnByValue: true,
+      });
```

Comment:
This needs a retry

Reply:
If wrapped in this way, a DeadlineError will appear after ~55 seconds of execution
```ts
const { result, exceptionDetails } = await retryDeadline(
  this.#celestial.Runtime.callFunctionOn({ ... }),
  this.#page.timeout, // 10_000
);
```
```ts
await element.evaluate(async () => {
  await new Promise((resolve) => setTimeout(resolve, 120_000));
})
```
RetryError: Retrying exceeded the maxAttempts (5).
Caused by DeadlineError: Deadline

maybe add as a `timeout` option and only use `deadline`?
```ts
interface ElementEvaluateOptions<T> {
  args: Readonly<T>;
  timeout?: number;
}

const evaluatePromise = this.#celestial.Runtime
  .callFunctionOn({
    functionDeclaration: func.toString(),
    objectId: object.objectId,
    arguments: args,
    awaitPromise: true,
    returnByValue: true,
  });

const { result, exceptionDetails } = await deadline(
  evaluatePromise,
  evaluateOptions?.timeout ?? this.#page.timeout,
);
```

Reply:
I think it's unrealistic that user code being evaluated takes more than 10 seconds to run. If it is, they can just change the timeout. I don't see this being an issue.

The real issue is actually the CDP protocol randomly never responds. This causes hangs that are impossible to reproduce. All CDP protocol runs should be retried and deadlined.